pyccel/ast/omp.py

```diff
@@ -0,0 +1,117 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+This file contain the OpenMP types.
+"""
+
+from .basic import Basic
+
+class OmpAnnotatedComment(Basic):
+
+    """Represents an OpenMP Annotated Comment in the code.
+
+    Parameters
+    ----------
+
+    txt: str
+        statement to print
+
+    combined: List (Optional)
+        constructs to be combined with the current construct
+
+    Examples
+    --------
+    >>> from pyccel.ast.omp import OmpAnnotatedComment
+    >>> OmpAnnotatedComment('parallel')
+    OmpAnnotatedComment(parallel)
+    """
+    _attribute_nodes = ()
+
+    def __init__(self, txt, combined=None):
+        self._txt = txt
+        self._combined = combined
+        super().__init__()
+
+    @property
+    def txt(self):
+        """Used to store clauses."""
+        return self._txt
+
+    @property
+    def combined(self):
+        """Used to store the combined construct of a directive."""
+        return self._combined
+
+    def __getnewargs__(self):
+        """Used for Pickling self."""
+
+        args = (self.txt, self.combined)
+        return args
+
+class OMP_For_Loop(OmpAnnotatedComment):
+    """ Represents an OpenMP Loop construct. """
+    def __init__(self, txt):
+        OmpAnnotatedComment.__init__(self, txt)
```

Comment:
Please use `super().__init__(txt)`. I spent some time trying to make this usage the standard wherever possible in pyccel. It ensures that the superclass `__init__` is called correctly even if the class is renamed, or a class is added to the inheritance tree

Reply:
Same comment for the other classes in this file